src/interfaces/protocols-configure.ts

```diff
@@ -71,4 +74,18 @@ export class ProtocolsConfigure extends Message<ProtocolsConfigureMessage> {
       types    : typesCopy,
     };
   }
+
+  public async authorize(tenant: string, messageStore: MessageStore): Promise<void> {
+    // if author is the same as the target tenant, we can directly grant access
+    if (this.author === tenant) {
+      return;
+    } else if (this.author !== undefined && this.authorizationPayload?.permissionsGrantId) {
+      await GrantAuthorization.authorizeGenericMessage(tenant, this, this.author, messageStore);
```

Comment:
Consider moving checks beyond `this.author === tenant` within `authorizeGenericMessage()`?
```suggestion
      await GrantAuthorization.authorizeGenericMessage(tenant, this, messageStore);
```

Reply:
Do you mind if I leave it as-is for now, maybe changing it in a later PR? I have an intuition that it'll be cleaner this way since other message types have different handling for anonymous messages. If it turns out to be messier as I implement grant-auth for other message types I'll move these checks into `authorizeGenericMessage()`.

I'm sorta following the pattern of `ProtocolAuthorization` where we check if protocol URI is present before even entering `ProtocolAuthorization.authorize()`.

Reply:
OK with it.